examples/shader/post_process_pass.rs

```diff
@@ -0,0 +1,410 @@
+        // Currently, every node in bevy uses this pattern, so it's safe to just copy it.
+        vec![SlotInfo::new(PostProcessNode::IN_VIEW, SlotType::Entity)]
+    }
+
+    // This will run every frame before the run() method
+    // The important difference is that `self` is `mut` here
+    fn update(&mut self, world: &mut World) {
+        // Since this is not a system we need to update the query manually.
+        // This is mostly boilerplate. There are plans to remove this in the future.
+        // For now, you can just copy it.
+        self.query.update_archetypes(world);
+    }
+
+    // Runs the node logic
+    // This is where you issue draw calls.
+    //
+    // This will run on every views the graph is running on. If you don't want your effect to run on every camera,
+    // you'll need to make sure you have a marker component to identify which camera should run the effect.
+    fn run(
+        &self,
+        graph_context: &mut RenderGraphContext,
+        render_context: &mut RenderContext,
+        world: &World,
+    ) -> Result<(), NodeRunError> {
+        // Get the entity of the view for the render graph where this node is running
+        let view_entity = graph_context.get_input_entity(PostProcessNode::IN_VIEW)?;
+
+        // We get the data we need from the world based on the view entity passed to the node.
+        // The data is the query that was defined earlier in the [`PostProcessNode`]
+        let Ok(view_target) = self.query.get_manual(world, view_entity) else {
+            return Ok(());
+        };
+
+        // Get the pipeline resource that contains the global data we need to create the render pipeline
+        let post_process_pipeline = world.resource::<PostProcessPipeline>();
+
+        // The pipeline cache is a cache of all previously created pipelines.
+        // It's required to avoid creating a new pipeline each frame, which is expensive due to shader compilation.
+        let pipeline_cache = world.resource::<PipelineCache>();
+
+        // Get the pipeline from the cache
+        let Some(pipeline) = pipeline_cache.get_render_pipeline(post_process_pipeline.pipeline_id) else {
+            return Ok(());
+        };
+
+        // Get the settings uniform binding
+        let settings_uniforms = world.resource::<ComponentUniforms<PostProcessSettings>>();
+        let Some(settings_binding) = settings_uniforms.uniforms().binding() else {
+            return Ok(());
+        };
+
+        // This will start a new "post process write"
+        //
+        // `source` is the "current" main texture. This will internally flip this
+        // [`ViewTarget`]'s main texture to the `destination` texture, so the caller
+        // _must_ ensure `source` is copied to `destination`, with or without modifications.
+        // Failing to do so will cause the current main texture information to be lost.
+        let post_process = view_target.post_process_write();
```

Comment:
At this point it is not clear what `source` nor `destination` are. I feel like something like this could be clearer.
```suggestion
        // This will start a new "post process write", obtaining two texture
        // views from the view target - a `source` and a `destination`.
        // `source` is the "current" main texture and you _must_ write into
        // `destination` because calling `post_process_write()` on the
        // [`ViewTarget`] will internally flip the [`ViewTarget`]'s main
        // texture to the `destination` texture. Failing to do so will cause
        // the current main texture information to be lost.
        let post_process = view_target.post_process_write();
```

Reply:
Good call, I just copied the docs from the post_process_write() directly.